core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -197,9 +197,14 @@ object GameStarter {
         val allMercantileResources = ruleset.tileResources.values.filter { it.unique == "Can only be created by Mercantile City-States" }.map { it.name }
         val unusedMercantileResources = Stack<String>()
         unusedMercantileResources.addAll(allMercantileResources.shuffled())
+        var numbersToTake = newGameParameters.numberOfCityStates
 
-        for (cityStateName in availableCityStatesNames.take(newGameParameters.numberOfCityStates)) {
+        for (cityStateName in availableCityStatesNames.take(numbersToTake)) {
             val civ = CivilizationInfo(cityStateName)
+            if (civ.cityStateType == CityStateType.Religious && !newGameParameters.religionEnabled){
+                numbersToTake++
+                continue
+            }
```

Comment:
Are you sure this works? Isn't the list for the for loop generated when entering the for loop, ie incrementing numbersToTake halfway through will have no effect?
It might be better to exclude the religious city states when first generating the `availableCityStatesNames`

Reply:
fixed